util-zk/src/main/scala/com/twitter/zk/coordination/ZkAsyncSemaphore.scala

```diff
@@ -262,12 +263,15 @@ class ZkAsyncSemaphore(zk: ZkClient, path: String, numPermits: Int, maxWaiters:
   }
 
   private[this] def numPermitsOf(node: ZNode): Future[Int] = {
-    node.getData() map { data =>
-      try {
-        new String(data.bytes, Charset.forName("UTF8")).toInt
-      } catch {
-        case err: NumberFormatException => -1
-      }
+    node.getData().transform {
+      case Return(data: ZNode.Data) =>
+        try {
+          Future(new String(data.bytes, Charset.forName("UTF8")).toInt)
```

Comment:
prefer Future.value, which doesn't require allocating an extra closure, especially if you're already in a try/catch block.